agenthub/langchains_agent/utils/memory.py

```diff
@@ -1,18 +1,42 @@
-from . import json
+import os
 
 import chromadb
-
 from llama_index.core import Document
 from llama_index.core.retrievers import VectorIndexRetriever
 from llama_index.core import VectorStoreIndex
 from llama_index.vector_stores.chroma import ChromaVectorStore
 
+from . import json
+
+embedding_strategy = os.getenv("LLM_EMBEDDING_MODEL", "openai")
+
+# TODO: More embeddings: https://docs.llamaindex.ai/en/stable/examples/embeddings/OpenAI/
+# There's probably a more programmatic way to do this.
+if embedding_strategy == "llama2":
+    from llama_index.embeddings.ollama import OllamaEmbedding
+    embed_model = OllamaEmbedding(
+        model_name="llama2",
+        base_url=os.getenv("LLM_EMBEDDING_MODEL_BASE_URL", "http://localhost:8000"),
+        ollama_additional_kwargs={"mirostat": 0},
+    )
+elif embedding_strategy == "local":
+    from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+    embed_model = HuggingFaceEmbedding(
+        model_name="BAAI/bge-small-en-v1.5"
+    )
```

Comment:
I have included another embedding model using azureopenai , raised PR for this branch, Let me know your review on it
thanks

Reply:
Merged!

I haven't used custom embeddings much. Can you help me understand if we should have separate env vars for base_url, api_key, etc? Or should they be the same as for the core LLM?

Reply:
Let's go with the smaller var for now--we can get more specific later